### PR TITLE
EOS-17487 "make devinstall" is broken: pip doesn't handle transitive dependencies well

### PR DESCRIPTION
**Pull request by supritshinde**
Problem: Transitive dependency issue (here due to idna)
Solution: Specifying the required version requirement in cfgen/requirement.txt and setup.py file so it resolves the dependency issues.

-----
[View rendered README.md](https://github.com/supritshinde/cortx-hare/blob/main/README.md)

### CODE REVIEW

**Comment by knekrasov** [comment]
@supritshinde Can you update the PR description? It states neither Problem nor Solution (while both are required).

File: hax/setup.py [hunk 1]
```diff
@@ -126,7 +126,8 @@ def get_motr_cflags():
     packages=find_packages(),
     setup_requires=['flake8', 'mypy', 'pkgconfig'],
     install_requires=[
-        'python-consul>=1.1.0', 'simplejson', 'aiohttp', 'click', 'dataclasses'
+        'python-consul>=1.1.0', 'simplejson', 'aiohttp', 'click',
```
Comment (knekrasov) [inline hax/setup.py:129]: Format the sources with [`yapf`](ssc-vm-2492.colo.seagate.com) please. I guess that would resolve Codacy issue here.

**Review by knekrasov — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -68,11 +68,14 @@ and health-checking mechanisms.
 
   * .. or from sources
     ```sh
+    
+    git clone --recursive https://github.com/Seagate/cortx-motr.git
+    cd cortx-motr
+
     git clone --recursive https://github.com/Seagate/cortx-motr.git motr
```
Comment (suvratjoshi) [inline README.md:75]: cloning command repeated with motr directory

File: README.md [hunk 3]
```diff
@@ -68,11 +68,14 @@ and health-checking mechanisms.
 
   * .. or from sources
     ```sh
+
```
Comment (suvratjoshi) [inline README.md:71]: probably this additional line needed to be removed (line 71)

Comment (supritshinde) [reply README.md:71]: Done!

File: cfgen/requirements.txt [hunk 4]
```diff
@@ -1,2 +1,3 @@
+idna<3,>=2.5
```
Comment (suvratjoshi) [inline cfgen/requirements.txt:1]: @knekrasov 
Do we need similar in file `provisioning\miniprov\requirements.txt`?

**Comment by supritshinde** [comment]
Invalid PR

**Pull request closed without merging**